### PR TITLE
Add ability to register nested types

### PR DESCRIPTION
**Pull request by LorienMan**
This should work now:
```
    // weaver: fuu = FuuTest21.Params<String>
```

For issue https://github.com/scribd/Weaver/issues/109

### CODE REVIEW

File: Sources/WeaverCodeGen/RegexUtils.swift [hunk 1]
```diff
@@ -32,9 +32,11 @@ enum Patterns {
     private static let equal = "\(spaces)=\(spaces)"
     private static let arrow = "\(spaces)<-\(spaces)"
     private static let name = "\\w+"
+    private static let typeNamePart = "\(name)(\\.\(name))*"
```
Comment (ibsh) [inline Sources/WeaverCodeGen/RegexUtils.swift]: I'm concerned that this won't allow arrays or dictionaries to reference nested types:
```
// weaver: foo <= [Bar.Baz]
// weaver: fuu <= [String: Bar.Baz]
```

Comment (LorienMan) [reply Sources/WeaverCodeGen/RegexUtils.swift]: oh, thank you!
I'll take a look

**Comment by LorienMan** [comment]
@ibsh I've managed to implement desired behavior, but I think that regexps should not be used to parse types.

For example you cannot use `SomeClass<[String: String]>` and `SomeClass<Dictionary<String, String>>` because Foundation regexp engine doesn't support recursion in patterns.

I'm thinking about some kind of a lexer for a robust solution.

File: Package.swift [hunk 2]
```diff
@@ -3,6 +3,9 @@ import PackageDescription
 
 let package = Package(
     name: "Weaver",
+    platforms: [
+       .macOS(.v10_13)
```
Comment (trupin) [inline Package.swift]: Is this intentional?

If yes, I'm concerned that this is limiting Weaver to be used on that specific version only.

Comment (LorienMan) [reply Package.swift]: Yes, it is. This line just moved deployment target from v10_10 (implicit) to v10_13 and resulted binary works great on the latest os version.
https://developer.apple.com/documentation/swift_packages/supportedplatform

10.13 is needed for regexp named groups:
```
NSTextCheckingResult
- (NSRange)rangeWithName:(NSString *)name API_AVAILABLE(macosx(10.13), ios(11.0), watchos(4.0), tvos(11.0));
```

**Comment by LorienMan** [comment]
@trupin @ibsh 
Hello guys! Is there anything else I should add to this PR?

**Comment by trupin** [comment]
Thanks for you contribution @LorienMan.

**Pull request merged**